### PR TITLE
cirrus: use fastvm for builds

### PR DESCRIPTION
**Pull request by Luap99**
Builds now take over 10 mins, given golang compilation is parallelized by default we can give more cores to speed it up.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by Luap99** [comment]
This seems to safe around 3mins, not a lot but at least something. I will look to see what parts of the build jobs are slow to see if there are some ways to make it faster because AFAICT most time does not seem to be spend compiling so I think there is some room in our docs/validation process.

**Comment by Luap99** [comment]
This should now safe around more than 4min of total CI time now.

Another PR on main with a Finished time of 10:22 from cirrus:
 image

This PR with a Finished time of 05:51 from cirrus:
image

**Comment by Luap99** [comment]
@edsantiago PTAL (low prio) I think this is good to review, of course this will fail CI until https://github.com/containers/podman/pull/24227 is merged then I will rebase and mark as ready.

**Review by lsm5 — APPROVED** [review]
LGTM

File: .cirrus.yml [hunk 1]
```diff
@@ -380,16 +377,18 @@ bindings_task:
...
         TEST_FLAVOR: bindings
     # N/B: This script depends on ${DISTRO_NV} being defined for the task.
     clone_script: &get_gosrc |
         cd /tmp
-        echo "$ARTCURL/Build%20for%20${DISTRO_NV}/repo/repo.tbz"
-        time $ARTCURL/Build%20for%20${DISTRO_NV}/repo/repo.tbz
-        time tar xjf /tmp/repo.tbz -C $GOSRC
+        echo "$ARTCURL/Build%20for%20${DISTRO_NV}/repo/repo.tar.zst"
+        time $ARTCURL/Build%20for%20${DISTRO_NV}/repo/repo.tar.zst
+        time tar --zstd -xf /tmp/repo.tar.zst -C $GOSRC
```
Comment (edsantiago) [inline .cirrus.yml:391]: From `info tar`:
>Reading compressed archive is even simpler: you don't need to specify
>any additional options as GNU ‘tar’ recognizes its format automatically.

I have a slight predilection for eliminating the `--zstd`, and just letting `tar` figure it out itself. WDYT?

Comment (Luap99) [reply .cirrus.yml:391]: I can try that and see if it works, the one thing that really messed me up is that we extract on macos as well and there is no gnu tar there so I am not sure if that can figure it out too. If not we end up with a case where one tar command has it and the other does not which might cause more confusion.

Comment (edsantiago) [reply .cirrus.yml:391]: Oh good point. I hate macs so much.

**Review by edsantiago — APPROVED** [review]
LGTM. I'm curious about the `tree_status` changes, especially how much we can rely on the state being what you describe, but will go along with it.

**Comment by edsantiago** [comment]
/lgtm

no need to squash. Thank you!

**Pull request merged**